client/implementation-vertx/src/main/java/io/smallrye/graphql/client/vertx/typesafe/HeaderBuilder.java

```diff
@@ -58,15 +51,38 @@ public MultiMap build() {
         return headers;
     }
 
-    private String resolveValue(Header header) {
+    private HeaderValue resolve(Header header) {
         if (!header.method().isEmpty()) {
             if (!header.constant().isEmpty())
                 throw new RuntimeException("Header with 'method' AND 'constant' not allowed: " + header);
-            return resolveMethodValue(header.method());
+            return new HeaderValue(resolveMethodValue(header.method()), header.name(), header.method());
         }
         if (header.constant().isEmpty())
             throw new RuntimeException("Header must have either 'method' XOR 'constant': " + header);
-        return header.constant();
+        if (header.name().isEmpty())
+            throw new RuntimeException("Missing header name for constant '" + header.constant() + "'");
+        return new HeaderValue(header.constant(), header.name(), null);
+    }
+
+    private HeaderValue resolve(ParameterInfo parameter) {
+        Header header = parameter.getAnnotations(Header.class)[0];
+        return new HeaderValue(parameter.getValue(), header.name(), parameter.getName());
+    }
+
+    private static class HeaderValue {
```

Comment:
I'd rather call this `HeaderDescriptor` or something, as it contains both the name and value?!

Reply:
fixed